src/features/hyperlinks.rs

```diff
@@ -73,7 +73,15 @@ fn format_osc8_hyperlink(url: &str, text: &str) -> String {
 }
 
 lazy_static! {
-    static ref COMMIT_LINE_REGEX: Regex = Regex::new("(.* )([0-9a-f]{40})(.*)").unwrap();
+    // The first group can be empty and needs to end with either a word-boundary or an ANSI escape code.
+    // This ensures all commit characters are consumed (from 7 to 40 chars).
+    static ref COMMIT_LINE_REGEX: Regex = Regex::new(format!(
+        "(.*(?:\\b|{osc}{style}))({commit})({suffix})",
+        osc = "\x1b\\[",
+        style = "[0-9;]*[mK]",
+        commit = "[0-9a-f]{7,40}",
+        suffix = ".*",
+    ).as_str()).unwrap();
```

Comment:
What do you think about keeping this regex simple and applying the function `ansi::strip_ansi_codes` to the input before passing it to the regex? The possible advantage of that approach is that it keeps all the ANSI-parsing logic in one place. That function should be quite fast as the underlying parsing logic comes from the vte terminal emulator project.

Reply:
Keeping the regex simple sounds interesting, wouldn't it remove colors from a custom `git log --format` though?

Reply:
Ah right, I forgot the matched parts of the regex were being relied on for the colored output. Ignore me then :)